ibis/backends/snowflake/__init__.py

```diff
@@ -106,7 +106,7 @@ def fetch_from_cursor(self, cursor, schema: sch.Schema) -> pd.DataFrame:
 
     def _get_schema_using_query(self, query):
         with self.begin() as bind:
-            result = bind.execute(query)
+            result = bind.execute(f"SELECT * FROM ({query}) t0 LIMIT 0")
```

Comment:
Why do we have to inline these `LIMIT 0` queries? Seems like having this in one place if we can is better than repeating it.

Reply:
`_get_schema_using_query` is called from both `con.sql` and `Table.sql` - the latter wasn't adding the `LIMIT 0` automatically, while the former was. While looking into this I noticed that:

- Some backends were already manually adding the `LIMIT 0` in their implementation
- Some backends had no need for a `LIMIT 0`

It felt cleaner to me to have `_get_schema_using_query` receive the raw query and handle it however the backend needs rather than relying on the caller to add a 0 limit automatically.

Reply:
Ah, ok. Yeah, that's cleaner.